homeassistant/components/matter/number.py

```diff
@@ -0,0 +1,159 @@
+"""Matter Number Inputs."""
+
+from __future__ import annotations
+from typing import Any
+from dataclasses import dataclass
+from chip.clusters import Objects as clusters
+from chip.clusters.Types import NullValue
+from matter_server.common.helpers.util import create_attribute_path_from_attribute
+
+from homeassistant.components.number import (
+    NumberEntity,
+    NumberEntityDescription,
+    NumberMode,
+)
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    Platform, 
+    EntityCategory,
+    UnitOfTime,
+    PERCENTAGE,
+)
+
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .entity import MatterEntity, MatterEntityDescription
+from .helpers import get_matter
+from .models import MatterDiscoverySchema
+
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Matter Number Input from Config Entry."""
+    matter = get_matter(hass)
+    matter.register_platform_handler(Platform.NUMBER, async_add_entities)
+
+
+@dataclass(frozen=True)
+class MatterNumberEntityDescription(NumberEntityDescription, MatterEntityDescription):
+    """Describe Matter Number Input entities."""
+
+
+class MatterNumber(MatterEntity, NumberEntity):
+    """Representation of a Matter Attribute as a Number entity."""
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        """Initialize the entity."""
+        super().__init__(*args, **kwargs)
+        self._attr_name =   self.entity_description.name + " EP " + str(self._endpoint.endpoint_id)
```

Comment:
I'm not a fan of this as many users will have no idea what "EP" means and most devices will only have a single endpoint. Also hardcoding it like this will make translation impossible.

Reply:
Maybe there's a better way, but I put that in to avoid two same-named numeric inputs. A particular example came up while I was working with the Inovelli switch which has two lighting endpoints so you get two OnLevel attributes. See image:  
<img width="235" alt="image" src="https://github.com/home-assistant/core/assets/15061942/86503866-25b7-45ab-8aa7-1854bbdf6f68">

Some thoughts ...

- I wanted something to distinguish between the two  and I thought that, at least with an EP designation, this could be explained in a user's manual for the device.  
- My "EP" approach also has the weakness that it doesn't clearly associate the numeric input with the specific control. Is it possible to relate the name of the numeric input to the control it relates to. I.e., if the user names the control "Living Room", is it possible for the numeric input to have that appended like ("OnLevel for Living Room"). That may be the least confusing if it was possible.
- Another approach that is likely better than using the "EP" would be to do what you did with events and see if there is a matching FixedLabel and append that value if a match can be found (or use the first label in FixedLabel as a default), but only append the "EP" if no match and the device has multiple endpoints.
- Longer term, I had thought there needs to be some kind of naming "helper" function as a more general naming function that does something like : See if there is a semantic value taglist and, if so, translate the semantic value to use in the entity name, or see if there is a FixedLabel and find one matching target labels  [X, Y, Z], or if no target label match in FixedLabel, use the first label as the name, or if nothing else, return an endpoint number.

Anyway, I agree with the point that that I put in isn't ideal, but less confusing than if there are duplicates with no way to distinguish.

Reply:
Opened a PR with a more generic approach to postfix the entity name, but only if the device actually has multiple endpoints:

https://github.com/home-assistant/core/pull/120038